### ISSUE TITLE
Sphinx v6.1.0+ raises ValueError("node line not found") during linkcheck

### ISSUE THREAD

**Issue by matthewfeickert** [sphinx-doc/sphinx#11119]
### Describe the bug

For `sphinx` `v6.1.0+` running `linkcheck` will fail with

```pytb
Exception occurred:
  File ".../lib/python3.10/site-packages/sphinx/util/nodes.py", line 278, in get_node_line
    raise ValueError("node line not found")
ValueError: node line not found
```

with the following traceback (for Sphinx `v6.1.0`): [sphinx-err-is6b_zcu.log](https://github.com/sphinx-doc/sphinx/files/10381008/sphinx-err-is6b_zcu.log)

For `sphinx<6.1.0` `linkcheck` will run as expected and pass without errors.

I don't have a minimal failing example at the moment, so I apologize but I am going to produce a fully reproducible full example, though I know that isn't preferable

### How to Reproduce

```console
$ docker run --rm -ti python:3.10 /bin/bash
root@25758314b85b:/# python -m venv venv && . venv/bin/activate
(venv) root@25758314b85b:/# python -m pip --quiet install --upgrade pip setuptools wheel
(venv) root@25758314b85b:/# apt-get update && apt-get install -y pandoc
(venv) root@25758314b85b:/# git clone https://github.com/scikit-hep/pyhf.git
(venv) root@25758314b85b:/# cd pyhf/
(venv) root@25758314b85b:/pyhf# python -m pip --quiet install --upgrade .[docs,test]
(venv) root@25758314b85b:/pyhf# python -m pip list | grep -i sphinx
nbsphinx                      0.8.11
Sphinx                        6.1.2
sphinx-click                  4.4.0
sphinx-copybutton             0.5.1
sphinx-issues                 3.0.1
sphinx-rtd-theme              0.5.1
sphinx-togglebutton           0.3.2
sphinxcontrib.applehelp       1.0.3
sphinxcontrib-bibtex          2.5.0
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
(venv) root@25758314b85b:/pyhf# cd docs
(venv) root@25758314b85b:/pyhf/docs# make linkcheck
...
reading sources... [100%] release-notes/v0.7.0                                                                                                                                        
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [ 91%] index                                                                                                                                                        
Exception occurred:
  File "/venv/lib/python3.10/site-packages/sphinx/util/nodes.py", line 280, in get_node_line
    raise ValueError("node line not found")
ValueError: node line not found
The full traceback has been saved in /tmp/sphinx-err-n3wxneib.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [Makefile:199: linkcheck] Error 2
(venv) root@25758314b85b:/pyhf/docs#
```

### Environment Information

```text
Platform:              linux; (Linux-5.15.0-56-generic-x86_64-with-glibc2.31)
Python version:        3.10.9 (main, Dec 21 2022, 18:59:22) [GCC 10.2.1 20210110])
Python implementation: CPython
Sphinx version:        6.1.2
Docutils version:      0.19
Jinja2 version:        3.1.2
Pygments version:      2.14.0
```

### Sphinx extensions

I haven't produced a minimal set yet, but I can work on it. [At the moment](https://github.com/scikit-hep/pyhf/blob/0c10f35ad22cc5d0f2176f66b657de3410e5e856/docs/conf.py#L40-L57) the extensions are

```python
extensions = [
    'sphinx.ext.autodoc',
    'sphinx.ext.autosummary',
    'sphinx.ext.coverage',
    'sphinx.ext.mathjax',
    'sphinx.ext.ifconfig',
    'sphinx.ext.viewcode',
    'sphinx.ext.githubpages',
    'sphinx.ext.intersphinx',
    'sphinxcontrib.bibtex',
    'sphinx.ext.napoleon',
    'sphinx_click.ext',
    'nbsphinx',
    'sphinx_issues',
    'sphinx_copybutton',
    'sphinx_togglebutton',
    'xref',
]
```

**Comment by francoisfreitag** [comment]
Narrowed down to the following index.rst:

```rst
.. image:: https://raw.githubusercontent.com/scikit-hep/pyhf/main/docs/_static/img/pyhf-logo-small.png
   :alt: pyhf logo
   :width: 320
   :align: center
```

**Comment by matthewfeickert** [comment]
Amazing! The Sphinx team is fantastically fast at resolving bugs when new releases come out and I can't thank you enough!

This has been fixed with Sphinx `v6.1.3` so I'm closing now.

**Comment by jfbu** [comment]
fixed at 31162a9b6 (released as `6.1.3`)
